it/controllers/pensions/annualAllowance/PstrSummaryControllerISpec.scala

```diff
@@ -0,0 +1,163 @@
+    val expectedButtonText = "Continue"
+    val expectedTitle = "Pension Scheme Tax Reference (PSTR) summary"
+    val expectedHeading = "Pension Scheme Tax Reference (PSTR) summary"
+    val change = "Change"
+    val remove = "Remove"
+    val pensionSchemeTaxReference = "Pension Scheme Tax Reference"
+    val expectedAddAnotherText = "Add another PSTR"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, Nothing]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY)
+  )
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        import Selectors._
+        import user.commonExpectedResults._
+
+        "render the 'PSTR Summary' page with pre-filled content" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropPensionsDB()
+            val viewModel = aPensionAnnualAllowanceViewModel.copy(pensionSchemeTaxReference = Some(Seq(pstr1, pstr2)))
+            insertCyaData(pensionsUserDataWithAnnualAllowances(viewModel), aUserRequest)
+            urlGet(fullUrl(pstrSummaryUrl(taxYearEOY)), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(expectedTitle)
+          h1Check(expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(pstr1, pstrSelector(1))
+          textOnPageCheck(pstr2, pstrSelector(2))
+          linkCheck(s"$change $change $pensionSchemeTaxReference $pstr1", changeLinkSelector(1),
+            PensionSchemeTaxReferenceController.show(taxYearEOY, Some(0)).url)
+          linkCheck(s"$change $change $pensionSchemeTaxReference $pstr2", changeLinkSelector(2),
+            PensionSchemeTaxReferenceController.show(taxYearEOY, Some(1)).url)
+          linkCheck(s"$remove $remove $pensionSchemeTaxReference $pstr1", removeLinkSelector(1), "#")
+          linkCheck(s"$remove $remove $pensionSchemeTaxReference $pstr2", removeLinkSelector(2), "#")
+          linkCheck(expectedAddAnotherText, addAnotherLinkSelector, PensionSchemeTaxReferenceController.show(taxYearEOY, None).url)
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+      }
+    }
+
+    "redirect to the annual allowance CYA page if there are no pension scheme tax references" should {
```

Comment:
As per previous comment I don't think this is right as you would not be able to add PSTRs if there were none.
In which case this test will need updating

Reply:
Updated tests to accommodate the changes above